lib/Sema/CSGen.cpp

```diff
@@ -1141,9 +1141,7 @@ namespace {
         auto appendingExprType = CS.getType(appendingExpr);
         auto appendingLocator = CS.getConstraintLocator(appendingExpr);
 
-        // Must be Conversion; if it's Equal, then in semi-rare cases, the 
-        // interpolation temporary variable cannot be @lvalue.
-        CS.addConstraint(ConstraintKind::Conversion, appendingExprType,
```

Comment:
@beccadax I don't suppose you recall the case this was handling? Doesn't seem to trip anything in the test suite